src/server/controllers/commentv2.controller.ts

```diff
@@ -156,3 +171,43 @@ export const toggleLockThreadDetailsHandler = async ({
     throw throwDbError(error);
   }
 };
+
+export const toggleHideCommentHandler = async ({
+  input,
+  ctx,
+}: {
+  input: ToggleHideCommentInput;
+  ctx: DeepNonNullable<Context>;
+}) => {
+  const { id: userId, isModerator } = ctx.user;
+  const { id, entityType } = input;
+
+  try {
+    const comment = await dbRead.commentV2.findFirst({
+      where: { id },
+      select: {
+        hidden: true,
+        userId: true,
+        thread: { select: { [entityType]: { select: { userId: true } } } },
+      },
+    });
+    if (!comment) throw throwNotFoundError(`No comment with id ${input.id}`);
+    if (
+      !isModerator &&
+      comment.userId !== userId &&
```

Comment:
Wouldn't this make it so that a non mod owner cannot hide other people's comment?